backend/src/services/event.service.ts

```diff
@@ -19,14 +20,40 @@ export class EventService {
     })
   }
 
-  // Retrieve events for a specific activity ID (refer to schema.prisma or diagram for relation)
-  async getActivityEvents(activity_id: string) {
-    // FInd many as one activity can have multiple events
-    return this.prisma.event.findMany({
+  async getMyEvents(user_id: number): Promise<EventWithNumUsers[]> {
+    const events = await this.prisma.event.findMany({
       where: {
-        // activity_id,
+        eventAssignments: {
+          // Find all events where there exists at least one event assignment for the user
+          some: {
+            user_id: user_id,
+          },
+        },
+      },
+      include: {
+        activity: true,
+        location: true,
+        eventAssignments: {
+          include: {
+            user: {
+              select: {
```

Comment:
```suggestion
      select: {
        id: true,
        start_time: true,
        end_time: true,
        on_call: true,
        activity: true,
        location: true,
        eventAssignments: {
          select: {
            id: true,
            user: {
              select: {
```

To match types, see big comment